extensions/liquibase/runtime/src/main/java/io/quarkus/liquibase/LiquibaseFactory.java

```diff
@@ -31,9 +34,23 @@ public Liquibase createLiquibase() {
         try (ClassLoaderResourceAccessor resourceAccessor = new ClassLoaderResourceAccessor(
                 Thread.currentThread().getContextClassLoader())) {
 
-            Database database = DatabaseFactory.getInstance()
-                    .findCorrectDatabaseImplementation(new JdbcConnection(dataSource.getConnection()));
-            ;
+            Database database;
+
+            if (config.username.isPresent() && config.password.isPresent()) {
+                AgroalDataSource agroalDataSource = (AgroalDataSource) dataSource;
```

Comment:
Using `Datasource#unwrap` may be a better option than explicit casting
```suggestion
                AgroalDataSource agroalDataSource = dataSource.unwrap(AgroalDataSource.class);
```